compiler/rustc_codegen_llvm/src/back/lto.rs

```diff
@@ -591,6 +591,18 @@ pub(crate) fn run_pass_manager(
     //      tools/lto/LTOCodeGenerator.cpp
     debug!("running the pass manager");
     unsafe {
+        if !llvm::LLVMRustHasModuleFlag(
+            module.module_llvm.llmod(),
+            "LTOPostLink".as_ptr().cast(),
+            11,
+        ) {
+            llvm::LLVMRustAddModuleFlag(
+                module.module_llvm.llmod(),
+                llvm::LLVMModFlagBehavior::Error,
+                "LTOPostLink\0".as_ptr().cast(),
+                1,
```

Comment:
Should this be 11?

Reply:
I believe the `HasModuleFlag` should be checking for `1`, but also adding this flag here seems super sketchy to me. This is a pretty internal thing to LTO AFAICT.

In that context, it would be great if this code had a comment with context as to why this is being done.

Reply:
In the mono repo, I found the value 1.

Reply:
The code here is based on the file tschuett linked above. In the LLVM repo that code is responsible to set up LTO. Adding this flag was missing from the Rust code that should emulate the code in `LTOCodeGenerator.cpp`.

The `11` argument in the `HasModuleFlag` is the size of the string passed to this function, not the value of the flag. The `1` argument in the `AddModuleFlag` function is the value of the module flag. LLVM doesn't need a string size here, because it expects a `\0` terminated string in this function.